pkg/controllers/serviceimport.go

```diff
@@ -114,6 +118,29 @@ func (r *ServiceImportReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, err
 	}
 	log.Info("created service")
+
+	if len(svcImport.Spec.IPs) == 0 {
```

Comment:
It's a small chance, but it seems like we want to verify the current state of the Service as well. As is, if the status update were to fail, we'd see the existing service on the next iteration and never set the LB IP. Since the updated Service controller ignores an import with a pre-existing IP, we wouldn't reconcile from the other side either

Reply:
I had the same thought as I was writing this initially :thinking: 

I agree with you, but I think that would require a larger refactor of this controller since we only reconcile the Service if it doesn't exist (based on line 93 - 97). So if the Service already exists, we would return early and never attempt to reconcile the status anyways. Was actually planning on adding the "update if already exists" logic for Services in a follow-up PR but I can do it here if you prefer.